antarest/study/business/allocation_management.py

```diff
@@ -0,0 +1,95 @@
+from typing import List
+
+from pydantic import BaseModel
+from starlette.responses import JSONResponse
+
+from antarest.study.business.utils import (
+    FormFieldsBaseModel,
+    execute_or_add_commands,
+)
+from antarest.study.model import Study
+from antarest.study.storage.storage_service import StudyStorageService
+from antarest.study.storage.variantstudy.model.command.update_config import (
+    UpdateConfig,
+)
+
+
+class AllocationField(FormFieldsBaseModel):
+    area_id: str
+    coefficient: float
+
+
+class AllocationFormFields(BaseModel):
+    allocation: List[AllocationField]
+
+
+class AllocationManager:
+    """
+    Provides functionality for getting and updating allocation fields for a given area
+    """
+
+    def __init__(self, storage_service: StudyStorageService) -> None:
+        self.storage_service = storage_service
+
+    def get_field_values(
+        self, study: Study, area_id: str, all_areas
+    ) -> JSONResponse:
+        file_study = self.storage_service.get_storage(study).get_raw(study)
+        allocation_data = file_study.tree.get(
+            f"input/hydro/allocation/{area_id}".split("/"), 2
```

Comment:
Used only one time, it may not be necessary to define a constant for it

Reply:
No it is also used in the setter

Reply:
Actually, I think it adds complexity because in Python, it is better to use a f-string rather than calling the `format` function. I'm not a fan of the DRY (Don't Repeat Yourself) concept in that specific case: a few repetitions don't hurt...
Also, we need a `list` (a sequence), so if we use a global constant, it would be better to have a list to avoid the `split` (which is not very nice).

Reply:
`f"input/hydro/allocation/{area_id}/[allocation]"`